README.md
# py-eagle-mqtt

Python3 based Docker for Eagle to MQTT reader

I have only ported this into a dockerfile, as well as made some changes to code for security or other purposes. All Original code is credit to [Ted Drain - TD22057](https://github.com/TD22057/T-Home).

This project utilizes the following tools:

[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
[![Libraries.io dependency status for GitHub repo](https://img.shields.io/librariesio/github/evanrich/py-eagle-mqtt)]()
[![Gitlab pipeline status (self-hosted)](https://img.shields.io/gitlab/pipeline/erichardson/py-eagle-mqtt/master?gitlab_url=https%3A%2F%2Fgitlab.evanrichardsonphotography.com)]()
[![Docker Pulls](https://img.shields.io/docker/pulls/evanrich/py-eagle-mqtt)]()
[![Docker Image Version (latest by date)](https://img.shields.io/docker/v/evanrich/py-eagle-mqtt)]()

## UPDATES:

2020-04-06: Moved Updates to a CHANGELOG.md file to clean this up.

## Usage

```
docker create \ 
  --name=py-eagle-mqtt \
  -e MQTT_BROKER_IP=<broker ip address> \
  -e MQTT_BROKER_PORT=<broker port number> \
  -e KEEPALIVE=<keepalive time in seconds> \
  -e MQTT_USER=<username>
  -e MQTT_PASSWORD=<password>
  -p 22042:22042 \
  evanrich/py-eagle-mqtt
```
&nbsp;

## Parameters

| Parameter | Function |
| :---: | --- |
| `-e MQTT_BROKER_IP` | The IP address of your MQTT Broker |
| `-e MQTT_BROKER_PORT` | The PORT your MQTT Broker listens on. Defaults to 1883 if not specified |

## Optional Parameters 
| Parameter | Function |
| :---: | --- |
| `-e KEEPALIVE` | Keepalive time in seconds. Defaults to 60 seconds if not specified |
| `-e MQTT_USER` | The User Name for your MQTT Broker if you use one. Defaults to None if not specified |
| `-e MQTT_PASSWORD` | The Password for your MQTT Broker if you use one. Defaults to None if not specified |

&nbsp;


snippet from my MQTT Broker showing container connecting and dissconnecting when I start/stop it:
```
{"pid":1,"hostname":"mqtt-867c776494-6rc57","name":"mosca","level":30,"time":1535955884767,"msg":"client connected","client":"","v":1}
{"pid":1,"hostname":"mqtt-867c776494-6rc57","name":"mosca","level":30,"time":1535955905640,"msg":"closed","client":"","v":1}
```

## Contributing

1. Fork it!
2. Create your feature branch: `git checkout -b my-new-feature`
3. Commit your changes: `git commit -am 'Add some feature'`
4. Push to the branch: `git push origin my-new-feature`
5. Submit a pull request :D



Original Readme as follows from creator, Ted Drain.  You can view the original repo this is based on here: https://github.com/TD22057/T-Home
---

T-Home Automation Software
==========================

A collection of scripts and utilities for various home automation projects.

- bin/  Command line tools
- conf/ Sample config files
- init.d/   Init.d style Linux start up scripts
- python/  Main scripting library
- systemd/  Systemd (latest Raspian) start up scripts
- upstart/  Upstart (Ubuntu 14.04) style start up scripts

Currently most of the scripts read data from various sources and
translate the data into JSON'ed dictionaries which get published to a
MQTT message broker.  


Rainforest Eagle Energy Monitor
-------------------------------

http://rainforestautomation.com/rfa-z109-eagle/

python/tHome/eagle contains code for reading data directly from an
Eagle energy monitor.  Use bin/tHome-eagle.py to start a small web
server and set the address as the "cloud provider" in the Eagle.  The
Eagle will publish energy data to the server which will converts it
into a message and publishes that as a MQTT messages.



CHANGELOG.md
## [1.7.3](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/compare/v1.7.2...v1.7.3) (2021-04-12)


### Bug Fixes

* Empty commit to get recent changes into build (last commit should've been labeled this) ([e513752](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/commit/e513752c46cc0b3beb93530f6549661d67196905))

## [1.7.2](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/compare/v1.7.1...v1.7.2) (2021-04-08)


### Bug Fixes

* Cleanup CHANGELOG.md since using new semantic-release format ([a72ac7a](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/commit/a72ac7a1df91cb2231e38e8bbceb5ccf9c93328a))
* fix asset info for git commit ([00c95c4](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/commit/00c95c42ea98921a1b2da9d2069d41e8e32f34f5))

## [1.7.1](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/compare/v1.7.0...v1.7.1) (2021-04-08)


### Bug Fixes

* refactor Dockerfile to improve build ([40ac8c5](https://gitlab.evanrichardsonphotography.com/erichardson/py-eagle-mqtt/commit/40ac8c5ada9b993f53b72dcc14b2d9b7cd183e2f))

## [1.6.10] - 2021-04-06

### Fixed

- Fix docker tagging issue

## [1.6.9] - 2021-04-06

### Fixed

- Remove release-cli section of CI file
- Add updated README.md

## [1.6.8] - 2021-04-06

### Changed

- Fixed slight issue with tagging

## [1.6.7] - 2021-04-06

### Changed

- Updated base image to Python 3.9.4-alpine3.13
- Moved changes to this file

### Removed

- Changelog from README.md

## [1.6.6] - 2021-04-03

### Added

- Added CI/CD pipeline to project

### Changed

- Moved project to my own personal Gitlab server for CI/CD pipelines as well as to run security scans on the container image and code

- Updated base image to Python 3.9.3-alpine3.12

## [Unversioned history]

- 2019-07-24: Rebase to python 3.7.4-alpine3.10, update bottle to 0.12.17

- 2019-04-12: Rebase to python 3.7.3-alpine3.9

- 2019-03-08: Rebase to python 3.7.2-alpine3.9, update bottle and astral versions to latest

- 2019-01-03: Rebase to python 3.7.2-alpine3.8, changed logging level to Info (should output to /var/log/tHome/eagle.log)

- 2018-09-10: Ported to Python3, Added pricing info.  Merged into master branch


Docker/src/conf/broker.py
import os
#===========================================================================
#
# MQTT message broker location.  Default broker location is port 1883
# for regular and 8883 for SSL.
#
#===========================================================================
host = os.getenv('MQTT_BROKER_IP', '192.168.1.20')
port = os.getenv('MQTT_BROKER_PORT', 1883)

# Keep alive time in seconds.  Client sends a ping if no other message
# is sent in this interval.
keepAlive = os.getenv('KEEPALIVE', 60)

#===========================================================================
#
# User name and password (strings) for broker log in.
#
#===========================================================================
user = os.getenv('MQTT_USER', None)
password = os.getenv('MQTT_PASSWORD', None)

#===========================================================================
#
# Secure connection options.  See the paho-mqtt docs for details.
#
#===========================================================================
# List of certificate files.
ca_certs = [
   ]

certFile = None
keyFile = None
